File: backend/src/schemas/analytics.py
from pydantic import BaseModel


class PricePredictionResponse(BaseModel):
    """Response schema for the predictive pricing endpoint."""

    crop_name: str
    county: str
    trend_slope: float
    predicted_price: float
    pi_low: float
    pi_high: float
    plain_language_insight: str

File: frontend/src/types/analytics.ts
export interface PricePrediction {
  crop_name: string;
  county: string;
  trend_slope: number;
  predicted_price: number;
  pi_low: number;
  pi_high: number;
  plain_language_insight: string;
}

export interface InventoryItem {
  id: string;
  farm_id: string;
  crop_name: string;
  quantity: number;
  unit: string;
  last_updated: string;
}
